### DISCUSSION TITLE
CrossCondition error in notice 23 ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#477] [Q&A] [answered]
Hi Lab !
in notice 23  BT-131(d)-Lot and BT-1311(d)-Lot are both mandatory : 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/6c27709d-989a-4a9b-b656-66d3cc3b081c)

but in validation they are alternative : 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/d5d9b76e-9ea8-43ce-a9d5-0f3ca0e2c1c4)

as in field.json : 

 "id" : "BT-131(d)-Lot",
    "parentNodeId" : "ND-SubmissionDeadline",
    "name" : "Deadline Receipt Tenders",
    "btId" : "BT-131",
    "xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingProcess/cac:TenderSubmissionDeadlinePeriod/cbc:EndDate",
    "xpathRelative" : "cbc:EndDate",
    "xsdSequenceOrder" : [ { "cbc:EndDate" : 4 } ],
    "type" : "date",
    "legalType" : "DATE",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "4", "5", "6", "10", "11", "12", "13", "14", "15", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "CEI", "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "7", "9", "16", "17", "18", "19", "20", "21", "22", "23", "24" ],
        "condition" : "{ND-LotTenderingProcess} ${BT-1311(d)-Lot is present}",
        "value" : true,
        "severity" : "ERROR"
      } ]
 

"id" : "BT-1311(d)-Lot",
    "parentNodeId" : "ND-ParticipationRequestPeriod",
    "name" : "Deadline Receipt Requests",
    "btId" : "BT-1311",
    "xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingProcess/cac:ParticipationRequestReceptionPeriod/cbc:EndDate",
    "xpathRelative" : "cbc:EndDate",
    "xsdSequenceOrder" : [ { "cbc:EndDate" : 4 } ],
    "type" : "date",
    "legalType" : "DATE",
    "repeatable" : {
      "value" : false,
      "severity" : "ERROR"
    },
    "forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "4", "5", "6", "8", "10", "11", "12", "13", "14", "15", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "CEI", "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "7", "9", "16", "17", "18", "19", "20", "21", "22", "23", "24" ],
        "condition" : "{ND-LotTenderingProcess} ${BT-131(d)-Lot is present}",
        "value" : true,
        "severity" : "ERROR"
      } ]

Bye
Renato

**Comment by YvesJo** [comment]
I guess the rules are initially evaluated (and with an empty form both fields appear as mandatory) and not reevaluated afterwards (hence not preventing the specification of the second field once the first one has been specified)

**Reply by rengit62** [reply]
The same situation about dates ( BT-131(d) and BT-1311(d) ) occurs in notice 16.
Both mandatory AND alternative ..I do not know of what you are writing ( "I guess ... )  What is the business need , are they alternative ? Must be at least one of them ? 
If I see XSD definition, under TenderingProcessType both TenderSubmissionDeadlinePeriod and ParticipationRequestReceptionPeriod are not mandatory, 

ND-SubmissionDeadline[repeatable=false,xpathRelative**=cac:TenderSubmissionDeadlinePeriod**]
				BT-131(d)-Lot[repeatable=false,xpathRelative=cbc:EndDate]
				BT-131(t)-Lot[repeatable=false,xpathRelative=cbc:EndTime]
			ND-ParticipationRequestPeriod[repeatable=false,xpathRelative=**cac:ParticipationRequestReceptionPeriod**]
				BT-1311(d)-Lot[repeatable=false,xpathRelative=cbc:EndDate]
				BT-1311(t)-Lot[repeatable=false,xpathRelative=cbc:EndTime]
			
 <xsd:complexType name="TenderingProcessType">
............................

 <xsd:element ref="**cac:TenderSubmissionDeadlinePeriod**"
                      minOccurs="0"
                      maxOccurs="1"/>
         <xsd:element ref="cac:InvitationSubmissionPeriod" minOccurs="0" maxOccurs="1"/>
         <xsd:element ref="cac:ParticipationInvitationPeriod" minOccurs="0" maxOccurs="1"/>
         <xsd:element ref="cac:ParticipationRequestReceptionPeriod"
                      minOccurs="0"c
                      maxOccurs="1"/>
        
so why you set both mandatories and alternative ?
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/c4a2e5b6-e12f-4aca-82f3-c148e31850ed)
Thanks and Regards
Renato

**Reply by YvesJo** [reply, marked as answer]
Hi,
These deadlines are alternative one from the other. Current rules enforce the existence of one or the other of these dates. It looks like wit the current set, not all situations are covered and an additional rule should be put in place to ensure that at least one of the two is specified. We still have to look into it.
